pandas/io/formats/style.py

```diff
@@ -575,75 +564,99 @@ def _translate(self):
 
         return d
 
-    def format(self, formatter, subset=None, na_rep: Optional[str] = None) -> Styler:
+    def format(
+        self,
+        formatter: Optional[
+            Union[Dict[Any, Optional[Union[str, Callable]]], str, Callable]
```

Comment:
can you type this at the top / in conftest.py (note that we can probably generally type these formatters across IO functions, but that's for another day)